.github/workflows/sp-deployment-pipeline.yml

```diff
@@ -0,0 +1,56 @@
+name: Deploy Sample Platform
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ master ]
+
+# permission can be added at job level or workflow level
+permissions:
+  id-token: write
+  contents: read # This is required for actions/checkout
+
+env:
+  DEPLOY_BRANCH: main
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_deployment:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Staging deployment with ssh commands using staging ssh key
+        uses: appleboy/ssh-action@master
+        with:
+          host: spdev.ccextractor.org
```

Comment:
We don't want the host to be public. We can use GitHub secrets here as well.

Reply:
But the host is public on the repository homepage description only, anyways keeping it a secret doesn't hurt though...